x-pack/plugin/core/build.gradle

```diff
@@ -39,6 +39,17 @@ dependencies {
     // security deps
     shadow 'com.unboundid:unboundid-ldapsdk:3.2.0'
     shadow project(path: ':modules:transport-netty4', configuration: 'runtime')
+    shadow(project(path: ':plugins:transport-nio', configuration: 'runtime')) {
+        // TODO: This should not be necessary if core is compile only
+        exclude group: "org.elasticsearch", module: "elasticsearch-core"
+        exclude group: "io.netty", module: "netty-buffer"
```

Comment:
We disable transitive dependencies for anything but `org.elasticsearch` so these excludes shouldn't be required either.

Reply:
transport-nit is under `org.elasticsearch`, that is why it gets transitive deps.